x-pack/solutions/security/plugins/security_solution/public/timelines/components/timeline/tabs/esql/index.tsx

```diff
@@ -59,16 +61,31 @@ export const DiscoverTabContent: FC<DiscoverTabContentProps> = ({ timelineId })
 
   const dispatch = useDispatch();
 
+  const { newDataViewPickerEnabled } = useEnableExperimental();
+  const { dataView: experimentalDataView } = useDataView(SourcererScopeName.detections);
+
   const { dataViewId } = useSourcererDataView(SourcererScopeName.detections);
 
-  const [dataView, setDataView] = useState<DataView | undefined>();
+  // eslint-disable-next-line prefer-const
+  let [dataView, setDataView] = useState<DataViewSpec | undefined>();
+
+  if (newDataViewPickerEnabled) {
+    dataView = experimentalDataView;
```

Comment:
I would just call `setDataview(experimentalDataView)` - if you're trying to avoid rerenders, then I would just use `useRef` instead.